packages/storage/src/cookies.ts

```diff
@@ -18,30 +18,36 @@ type CookieProperties = {
   sameSite?: "None" | "Lax" | "Strict";
 };
 
-const cookiePropertyKeys = [
-  "domain",
-  "expires",
-  "path",
-  "secure",
-  "httpOnly",
-  "maxAge",
-  "sameSite",
-] as const;
+type CookieProperties = { [key in keyof CookiePropertyTypes]: CookiePropertyTypes[key]; }
+
+const cookiePropertyMap = {
+  "domain": "Domain",
+  "expires": "Expires",
+  "path": "Path",
+  "secure": "Secure",
+  "httpOnly": "HttpOnly",
+  "maxAge": "Max-Age",
+  "sameSite": "SameSite",
+} as const;
 
 function serializeCookieOptions(options?: CookieOptions) {
   if (!options) return "";
-  return Object.entries(options)
+  const result = Object.entries(options)
     .map(([key, value]) => {
-      if (key === "maxAge") {
-        key = "max-age";
-      }
-      return value instanceof Date
-        ? `; ${key}=${value.toUTCString()}`
-        : typeof value === "boolean"
-          ? `; ${key}`
-          : `; ${key}=${value}`;
-    })
-    .join("");
+      const serializedKey = cookiePropertyMap[key as keyof CookiePropertyTypes];
+      if (!serializedKey)
+        return undefined;
+
+      if (value instanceof Date)
+        return `${serializedKey}=${value.toUTCString()}`;
+      if (typeof value === "boolean")
+        return value ? `${serializedKey}` : undefined;
+      return `${serializedKey}=${value}`;
+    });
+
+  return result.length != 0
+    ? `; ${result.join("; ")}`
```

Comment:
This would leave "undefined" in the result whenever either the serializedKey is missing or there is a key with the value false. Maybe filter(Boolean) the result.

Reply:
You're right, thank's I assumed that join will ignore `undefined` values